### PR TITLE
apply review changes

### LINKED ISSUE #2: remove pausability

### LINKED ISSUE #3: add functions to change` _baseURL` from admin

### LINKED ISSUE #4: remove freeze function (transfers are frozen by default and unfrozen only once)

### LINKED ISSUE #5: simplify ownership to only 1 role, maybe remove OZ and use Major implementation

### LINKED ISSUE #7: add `depositAddress` to constructor

### PR DESCRIPTION
**Pull request by MarcoWorms**
closes #2, closes #3, closes #4, closes #5, closes #7 
- #2 
- #3
- #4
- #5
- #7

### CODE REVIEW

File: YFUtechne.sol [hunk 1]
```diff
@@ -2,54 +2,42 @@
...
 
-    bytes32 public constant PAUSER_ROLE = keccak256("PAUSER_ROLE");
-    bytes32 public constant TRANSFER_FREEZER_ROLE = keccak256("TRANSFER_FREEZER_ROLE");
     Counters.Counter private _tokenIdCounter;
 
     uint256 public MAX_SUPPLY = 10;
     uint256 public PRICE = 1 ether;
     address payable public depositAddress;
     bool public transfers_frozen = true;
+    string public uri = "https://ipfs.io/ipfs/QmSt5CVksdLfvaDCHwPvTGtXs7YrSGcQ2raE3M9nXPifZH/";
```
Comment (Majorfi) [inline YFUtechne.sol]: I would not set it by default. Best to leave it empty, and populate with constructor maybe

File: YFUtechne.sol [hunk 2]
```diff
@@ -2,54 +2,42 @@
...
 
-    constructor() ERC721("YFU Techne", "YFU_1") {
-        depositAddress = payable(msg.sender);
-        _grantRole(DEFAULT_ADMIN_ROLE, msg.sender);
-        _grantRole(PAUSER_ROLE, msg.sender);
-        _grantRole(TRANSFER_FREEZER_ROLE, msg.sender);
+    constructor(address payable adminAddress) ERC721("YFU Techne", "YFU_1") {
+        depositAddress = adminAddress;
+        _grantRole(DEFAULT_ADMIN_ROLE, adminAddress);
     }
```
Comment (Majorfi) [inline YFUtechne.sol]: Maybe add the value for the URI here so we can deploy and set directly based on IPFS deployment

**Review by Majorfi — CHANGES_REQUESTED** [review]
Hello @MarcoWorms !
Thank you for the edits, here is a basic review from a non-expert 😬

File: YFUtechne.sol (hunk 2 above)
Comment (MarcoWorms) [reply YFUtechne.sol]: totally agree with the change!

**Comment by MarcoWorms** [comment]
merged, any other change requests just open an issue!

**Pull request merged**